pkg/kubelet/kubelet_pods.go

```diff
@@ -2088,6 +2099,8 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 			if ephemeralStorage, found := container.Resources.Limits[v1.ResourceEphemeralStorage]; found {
 				limits[v1.ResourceEphemeralStorage] = ephemeralStorage.DeepCopy()
 			}
+
+			convertCustomResources(container.Resources.Limits, limits)
```

Comment:
Alternatively: just deep-copy the list, and then special-case the memory & cpu limits. E.g.

```go
container.Resources.Limits = limits.DeepCopy()
if cStatus.Resources != nil && cStatus.Resources.CPULimit != nil {
  // ...
```